nemo/collections/nlp/modules/common/text_generation_strategy.py

```diff
@@ -268,10 +269,23 @@ def __init__(self, model):
     def clip_max_len(self, maxlen: int) -> int:
         """clip the max len based on the LM model max sequence length"""
 
-        # for positional embedding types that allow length extrapolation, don't clip the max length
         if self.model.cfg.get("position_embedding_type", "learned_absolute") == "learned_absolute":
             if maxlen > self.model.cfg.encoder_seq_length + 1:
                 maxlen = self.model.cfg.encoder_seq_length + 1
+        else:
+            # for positional embedding types that allow length extrapolation, clip based on the max_seq_length
+            # which is set by the user.
+            if self.model.trainer.state.fn.value == "fit":
+                max_limit = self.model.cfg.data.train_ds.max_seq_length
+            elif self.model.trainer.state.fn.value == "validate":
+                max_limit = self.model.cfg.data.train_ds.max_seq_length
```

Comment:
shouldn't we ideally get this from validation_ds?

Reply:
oops. will  fix.